client.go

```diff
@@ -28,8 +28,11 @@ type Client struct {
 	// UserAgent configures the User-Agent HTTP header for requests. Defaults to "tailscale-client-go".
 	UserAgent string
 	// APIKey allows specifying an APIKey to use for authentication.
-	// To use OAuth Client credentials, construct an [http.Client] using [OAuthConfig] and specify that below.
+	// To use OAuth Client credentials, specify OAuthCredentials instead.
 	APIKey string
+	// OAuthCredentials specifies credentials for authenticating with OAuth. If provided,
+	// APIKey is ignored.
+	OAuthCredentials *OAuthCredentials
```

Comment:
Combining all of the OAuth credentials into a single type so that it's just a single configuration field on the `Client` is nice because it avoids confusion about which combinations of fields are required for what.